flask_lesson_intro/main.py

```diff
@@ -0,0 +1,38 @@
+from flask import Flask, render_template, abort
+from json.decoder import JSONDecodeError
+from utils import get_data
+
+app = Flask(__name__)
+
+
+@app.route('/')
+def get_home_page():
+    return render_template("home.html")
+
+
+@app.route('/author')
+def get_author_page():
+    return render_template("author.html")
+
+
+@app.route('/<item>')
+def get_item_page(item):
+    try:
+        data = get_data()
+    except (FileNotFoundError, JSONDecodeError):
```

Comment:
move catching exceptions on get_data function

Reply:
and when if you catch exception set data as `{}`
Here just check `if not data: 404`